src/transpiling/mod.rs

```diff
@@ -205,6 +220,14 @@ impl crate::swc::common::errors::Emitter for DiagnosticCollector {
   }
 }
 
+#[derive(Debug, Error)]
+pub enum FoldProgramError {
+  #[error(transparent)]
+  ParseDiagnostics(#[from] ParseDiagnosticsError),
+  #[error(transparent)]
+  Swc(#[from] SwcFoldDiagnosticsError),
+}
```

Comment:
We should switch this to actually use the diagnostics in the future. This is a very rough/sloppy first pass.

Reply:
SGTM